packages/docusaurus-theme-classic/src/index.js

```diff
@@ -19,39 +19,47 @@ const ContextReplacementPlugin = requireFromDocusaurusCore(
 // Need to be inlined to prevent dark mode FOUC
 // Make sure that the 'storageKey' is the same as the one in `/theme/hooks/useTheme.js`
 const storageKey = 'theme';
-const noFlash = (defaultDarkMode) => `(function() {
+const noFlash = ({defaultDarkMode = false, respectUserPreference = false}) => {
+  return `(function() {
   var defaultDarkMode = ${defaultDarkMode};
+  var respectUserPreference = ${respectUserPreference};
```

Comment:
It would be better to  more explicitly describe in the option itself that what exactly is this preference. `respectUserDarkModeSystemPreference` (being elaborate here)

Reply:
https://drafts.csswg.org/mediaqueries-5/#prefers-color-scheme

I don't like the idea to have `dark mode` in the attribute name. As it's specified, the preferred color scheme may be extended in the future to other values like "sepia" (wouldn't be surprised to see new values being created for color blind people in the future).

Also not fan of "system". It's not part of the spec where this preference is configured exactly. It can be your OS, it can be your browser, it can be a browser plugin...